internal/datanode/compaction_executor_test.go

```diff
@@ -124,6 +123,7 @@ func newMockCompactor(isvalid bool) *mockCompactor {
 		ctx:     ctx,
 		cancel:  cancel,
 		isvalid: isvalid,
+		done:    make(chan struct{}, 1),
```

Comment:
use a unbuffered channel will be better as we don't need to type "1" many times

Reply:
with unbuffered channel, if call `done<- struct{}{}` after `<-done`, will cause deadlock

Reply:
> with unbuffered channel, if call `done<- struct{}{}` after `<-done`, will cause deadlock

We can just close this channel, not to write anything into it